cpp/src/arrow/util/spaced_sse_codegen.py

```diff
@@ -0,0 +1,185 @@
+#!/bin/python
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Generate the mask of lookup table for spaced compress and expand
+# Modified from original source:
+# https://github.com/lemire/simdprune/blob/master/scripts/epi32.py
+# https://github.com/lemire/simdprune/blob/master/scripts/epi8.py
+# The original copyright notice follows.
+
+# This code is released under the
+# Apache License Version 2.0 http://www.apache.org/licenses/.
+# (c) Daniel Lemire
+
+# Usage: python3 spaced_sse_codegen.py > spaced_sse_generated.h
+
+
+def print_mask_expand_bitmap(width, length):
```

Comment:
Is it possible to have an explanation for the data generation?

Reply:
Will add